File: leetcode/2748-number-of-beautiful-pairs/countBeautifulPairs.go
package leetcode

func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func countBeautifulPairs(nums []int) int {
	var count = 0
	var length = len(nums)

	for i := 0; i < length; i++ {
		var first = nums[i]
		for first >= 10 {
			first = first / 10
		}

		for j := i + 1; j < length; j++ {
			var end = nums[j] % 10
			if gcd(first, end) == 1 {
				count++
			}
		}
	}

	return count
}

File: leetcode/2748-number-of-beautiful-pairs/countBeautifulPairs.rs
use crate::Solution;

impl Solution {
    pub fn count_beautiful_pairs(nums: Vec<i32>) -> i32 {
        let mut count = 0;
        let length = nums.len();

        for i in 0..length {
            let mut first = nums[i];
            while first >= 10 {
                first = first / 10
            }

            for j in i + 1..length {
                let end = nums[j] % 10;

                if Self::gcd(first, end) == 1 {
                    count = count + 1
                }
            }
        }

        return count;
    }

    fn gcd(mut a: i32, mut b: i32) -> i32 {
        while b != 0 {
            let temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}
